table/data_block_hash_index.h

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2011-present, Facebook, Inc. All rights reserved.
+//  This source code is licensed under both the GPLv2 (found in the
+//  COPYING file in the root directory) and Apache 2.0 License
+//  (found in the LICENSE.Apache file in the root directory).
+
+#pragma once
+
+#include <string>
+#include <vector>
+
+namespace rocksdb {
+
+// The format of the datablock hash map is as follows:
+//
+// B B B ... B IDX NUM_BUCK MAP_START
+//
+// NUM_BUCK: Number of buckets, which is the length of the IDX array.
+//
+// IDX:      Array of offsets of the index hash bucket (relative to MAP_START).
+//
+// B:        B = bucket, an array consisting of a list of restart index, and the
+//           second hash value as tag.
+//           We do not have to store the length of individual buckets, as they
+//           are delimited by the next bucket offset.
+//
+// MAP_START: the starting offset of the datablock hash map.
+//
+// Each bucket B has the following structure:
+// [TAG RESTART_INDEX][TAG RESTART_INDEX]...[TAG RESTART_INDEX]
+// where TAG is the hash value of the second hash funtion.
```

Comment:
Lets explain here clearly how to locate a key/value by giving a example. 
Can you also provide more information here about why you decided to use as second hash function? Put a reference to the paper too from which this double-hashing idea is taken, so that the original authors get the credit.